internal/go_repository.bzl

```diff
@@ -147,7 +147,13 @@ def _go_repository_impl(ctx):
             print("fetch_repo: " + result.stderr)
 
     if generate:
-        # Build file generation is needed
+        # Build file generation is needed. Populate Gazelle directive at root build file
+        ctx.file(
```

Comment:
We should only write this if `ctx.attr.gazelle_directives` is set.

Ideally we would append to the file rather than replacing it if it already exists. I don't think there's a simple way to do that though. Perhaps setting this attribute should be an error if the file exists.

Reply:
If `build_file_generation = "auto"` and build files exist in the repo, Gazelle will not run and no Gazelle directive is needed. We can just ignore them. If `build_file_generation = "on"`, `go_repository` will override the existing build files if there is any. Overriding the root build file with Gazelle directives feels consistent with the behavior. Do you think it necessary to report an error for this case?